### PR TITLE
feat: connect moderate page to api

### PR DESCRIPTION
**Pull request by ilasw**
## Changes
- Fix moderate page heading navigation for users (removed settings page)
- Connect Infinite query for post moderation page
- Approve/Reject flow with real GraphQL APIs + small refactors to it

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### Preview domain
https://feat-moderate-page-api.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/squads/SquadTabs.tsx [hunk 1]
```diff
@@ -25,10 +27,10 @@ export function SquadTabs({
       : SquadTab.PendingPosts;
 
     return {
-      [SquadTab.Settings]: `${host}/edit`,
+      ...(showSettings && { [SquadTab.Settings]: `${host}/edit` }),
       [title]: `${host}/moderate`,
     };
-  }, [handle, pendingCount]);
+  }, [handle, pendingCount, showSettings]);
```
Comment (ilasw) [inline packages/shared/src/components/squads/SquadTabs.tsx:33]: Before this users can see the `Settings` tab, now we check outside if user is moderator and pass `showSettings`.

File: packages/shared/src/hooks/squads/useSquadPostModeration.ts [hunk 2]
```diff
@@ -45,51 +48,95 @@ export const rejectReasons: { value: PostModerationReason; label: string }[] = [
...
     if (onSuccessRef.current) {
       onSuccessRef.current(null);
       onSuccessRef.current = undefined;
     }
   };
 
+  const invalidateQueries = () => {
+    Promise.all([
+      queryClient.invalidateQueries({ queryKey: listQueryKey }),
+      queryClient.invalidateQueries({ queryKey: squadQueryKey }),
```
Comment (ilasw) [inline packages/shared/src/hooks/squads/useSquadPostModeration.ts:93]: need to invalidate this query for `moderationPostCount`

File: packages/webapp/pages/squads/[handle]/moderate.tsx [hunk 3]
```diff
@@ -31,23 +33,29 @@ interface ModerateSquadPageProps {
...
-          variant={ButtonVariant.Tertiary}
+          href={`/squads/${squad.handle}`}
           icon={<ArrowIcon className="-rotate-90" />}
+          tag="a"
+          variant={ButtonVariant.Tertiary}
         />
-        <PageHeaderTitle className="typo-title3">
-          Squad settings
-        </PageHeaderTitle>
+        <PageHeaderTitle className="typo-title3">Squad</PageHeaderTitle>
```
Comment (ilasw) [inline packages/webapp/pages/squads/[handle]/moderate.tsx:52]: Changed to make it compatible with user navigation

File: packages/shared/src/hooks/squads/useSquadPostModeration.ts [hunk 4]
```diff
@@ -45,51 +48,85 @@ export const rejectReasons: { value: PostModerationReason; label: string }[] = [
...
-      onSuccessWrapper();
+      invalidateQueries();
+    },
+    onError: (_, variables) => {
+      if (variables.postIds.length > 50) {
+        displayToast(
+          'Failed to approve post(s). Please approve maximum 50 posts at a time',
+        );
+        return;
+      }
```
Comment (sshanzel) [inline packages/shared/src/hooks/squads/useSquadPostModeration.ts:110]: Any particular reason we have this? Was it a product's requirement?

File: packages/shared/src/graphql/squads.ts [hunk 5]
```diff
@@ -609,8 +608,72 @@ export interface SourcePostModeration extends Partial<PostRequestContent> {
...
+  }
+  ${SOURCE_POST_MODERATION_FRAGMENT}
+`;
+
+export const SQUAD_MODERATE_POST_MUTATION = gql`
+  mutation ModerateSourcePost(
+    $postIds: [ID]!
+    $status: String
+    $sourceId: ID!
+    $rejectionReason: String
```
Comment (sshanzel) [inline packages/shared/src/graphql/squads.ts:661]: Reason should be required when rejecting if I'm not mistaken.

```suggestion
    $rejectionReason: String!
```

File: packages/shared/src/components/squads/moderation/SquadModerationList.tsx [hunk 6]
```diff
@@ -42,65 +48,97 @@ interface SquadModerationListProps {
...
+    );
+  }
 
+  if (squad.currentMember.role === SourceMemberRole.Member) {
     return (
       <SquadEmptyScreen
         Icon={VIcon}
         title="All done!"
-        description="All caught up! There are no posts waiting for your review right now."
+        description="All caught up! No posts are pending" // TODO:: MI-597 - check with product what to show here
```
Comment (sshanzel) [inline packages/shared/src/components/squads/moderation/SquadModerationList.tsx:72]: I think we can just use what we have here. If they have objections, we can always update the copy.

```suggestion
        description="All caught up! No posts are pending"
```

File: packages/shared/src/hooks/squads/useSquadPostModeration.ts (hunk 4 above)
Comment (ilasw) [reply packages/shared/src/hooks/squads/useSquadPostModeration.ts:110]: Talked to @capJavert  and he proposed to put the limit on BE + a logger to make sure to catch if this case is a common one. I think this is a good catch considering how rare is to approve 50+ posts at once 👀

File: packages/shared/src/components/squads/moderation/SquadModerationList.tsx (hunk 6 above)
Comment (ilasw) [reply packages/shared/src/components/squads/moderation/SquadModerationList.tsx]: Done ✔️

**Review by sshanzel — APPROVED** [review]
Looks good to me! 🚀

**Pull request merged**